### PR TITLE
Add reusable DRT build

### PR DESCRIPTION
**Pull request by khieta**
*Issue #, if available:*
[cedar#488](https://github.com/cedar-policy/cedar/issues/488)

*Description of changes:*
* Add yaml file to build and test DRT that can be used in upstream CI.
* Delete the `cedar` submodule. The appropriate `cedar` version will be pulled in by CI.
* Get rid of the `build.sh` script. The issue of a reusable build is addressed by the yaml file from the first bullet.
* Add instructions on how to use the Dockerfile.
* Add CI to check that the Dockerfile builds.

### CODE REVIEW

**Review by john-h-kastner-aws — APPROVED** [review]

**Comment by khieta** [comment]
Ready for (re-)review!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,44 +40,17 @@ jobs:
...
-        shell: bash
-        run: source ~/.profile && ./build.sh
-
-  run-integration-tests:
-    uses: ./.github/workflows/run_integration_tests_reusable.yml
-    with:
-      cedar_spec_ref: ${{ github.ref }}
\ No newline at end of file
+      - name: Build Docker image
+        run: docker build .
```
Comment (cdisselkoen) [inline .github/workflows/ci.yml:56]: Not sure if we should consider using the Docker reusable action, like I did in `cedar-examples`.  (I wasn't aware you can just run `docker build` haha; now I'm not sure which is better.)

File: README.md [hunk 2]
```diff
@@ -7,29 +7,42 @@ This repository contains the formalization of Cedar and infrastructure for perfo
...
 * `cd cedar-lean && lake build Cedar`
 
-To build the DRT framework:
+### DRT framework
+
+The simplest way to build our DRT framework is to use the included Dockerfile:
+
+```bash
+docker build . -t cedar_drt # ~10 minutes
+docker run -it cedar_drt
```
Comment (cdisselkoen) [inline README.md:25]: Might want to suggest `--rm`, which I think is a very sensible default when using `-it`

**Review by cdisselkoen — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (khieta) [reply .github/workflows/ci.yml]: The reusable action makes more sense, especially if we want to publish the image somewhere later.

File: Dockerfile [hunk 3]
```diff
@@ -25,9 +24,14 @@ FROM prepare AS build
 ENV CEDAR_SPEC_ROOT=/opt/src/cedar-spec
 COPY . $CEDAR_SPEC_ROOT
 
-# Build DRT
+# Clone `cedar` repository
 WORKDIR $CEDAR_SPEC_ROOT
-RUN source /root/.profile && ./build.sh
+RUN git clone https://github.com/cedar-policy/cedar
```
Comment (john-h-kastner-aws) [inline Dockerfile:29]: Minor improvement to image build time and size
```suggestion
RUN git clone --depth 1 https://github.com/cedar-policy/cedar
```

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**